cot/src/common_types.rs

```diff
@@ -695,6 +698,78 @@ impl Display for Email {
     }
 }
 
+#[derive(Debug, Error)]
+pub(crate) enum DateTimeWithOffsetConversionError {
+    #[error("nanoseconds out of range")]
+    NanosecondsOutOfRange,
+    #[error("offset not in valid range")]
+    InvalidOffset,
+    #[error("datetime out of range for conversion")]
+    TimestampOutOfRange,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
+pub(crate) struct DateTimeWithOffsetAdapter(DateTime<FixedOffset>);
```

Comment:
Not quite sure about the name and location of this struct, but I had to abstract this logic since it's now being referenced in multiple locations. Any suggestions?